src/components/DataSession.vue

```diff
@@ -12,43 +22,15 @@ const props = defineProps({
 	}
 })
 
-function addOperation(operationDefinition) {
-	// Create data operation in this session with input parameters specified from wizard
-	fetch('http://127.0.0.1:8000/api/datasessions/' + props.data.id + '/operations/', {
-		'method': 'POST',
-		'body': JSON.stringify(operationDefinition),
-		'headers': {
-			'Authorization': 'Token 123456789abcdefg',
-			'Content-type': 'application/json; charset=UTF-8'
-		}
-	})
-		.then(response => response.json())
-		.then(function() {emit('reloadSession')})
-		.catch(error => console.error('Error:', error))
+async function addOperation(operationDefinition) {
+	const url = dataSessionsUrl + props.data.id + '/operations/'
+	await fetchApiCall({url: url, method: 'POST', body: operationDefinition, headers: authHeaders, successCallback: emit('reloadSession'), failCallback: handleError})
```

Comment:
I _think_ we have to `JSON.stringify(operationDefinition)`

Reply:
I tested my changes to make sure they work before putting up any PRs. It actually caught me by surprise cause in an earlier commit I didn't change it and I got an error. But it turns out the fetchapi handles it already with this -> 'body: body ? JSON.stringify(body) : null'

Reply:
oh yeah! good eye